lnwallet/btcwallet/btcwallet.go

```diff
@@ -656,14 +656,17 @@ func (b *BtcWallet) SubscribeTransactions() (lnwallet.TransactionSubscription, e
 // it has fully synced to the current best block in the main chain.
 //
 // This is a part of the WalletController interface.
-func (b *BtcWallet) IsSynced() (bool, error) {
+func (b *BtcWallet) IsSynced() (bool, int64, error) {
+	var err error
 	// Grab the best chain state the wallet is currently aware of.
+	// also grab the timestamp of the best block to return for determining
+	// sync progress
 	syncState := b.wallet.Manager.SyncedTo()
+	walletBestBlock, err := b.chain.GetBlockHeader(&syncState.Hash)
```

Comment:
Check out the bottom of this function: as is, we'd fetch the header for a _second_ unnecessarily. Also we can't directly reference the `chain` here as we need to fetch the header from the backend.

Reply:
That was my thought - to use the header that was already being fetched at the end of the function. That value seems not to change as the sync progresses. Which implies to me it's the header at the tip of the, i.e. btcd chain  and not the header at the tip of the wallet's chain. I just double checked and that does indeed seem to be the case.

Am I overlooking something?